firewood/tests/v2api.rs

```diff
@@ -0,0 +1,69 @@
+use std::{error::Error, path::PathBuf, sync::Arc};
+
+use firewood::{
+    db::{BatchOp, Db as PersistedDb, DbConfig, DbError, WalConfig},
+    v2::api::{Db, DbView, Proposal},
+};
+
+#[tokio::test(flavor = "multi_thread")]
+async fn smoke() -> Result<(), Box<dyn Error>> {
+    let cfg = DbConfig::builder()
+        .meta_ncached_pages(1024)
+        .meta_ncached_files(128)
+        .payload_ncached_pages(1024)
+        .payload_ncached_files(128)
+        .payload_file_nbit(16)
+        .payload_regn_nbit(16)
+        .wal(
+            WalConfig::builder()
+                .file_nbit(15)
+                .block_nbit(8)
+                .max_revisions(10)
+                .build(),
+        )
+        .truncate(true)
+        .build();
+    let db = Arc::new(testdb(cfg).await?);
+    let empty_hash = db.root_hash().await?;
+    assert_ne!(empty_hash, [0; 32]);
+
+    // insert a single key/value
+    let batch_put = BatchOp::Put {
+        key: b"smoke",
+        value: b"test",
+    };
+    let proposal: Arc<firewood::db::Proposal> = db.propose(vec![batch_put]).await?.into();
+    proposal.commit().await?;
+
+    // ensure the latest hash is different
+    let latest = db.root_hash().await?;
+    assert_ne!(empty_hash, latest);
+
+    // fetch the view of the latest
+    let view = db.revision(latest).await.unwrap();
+
+    // check that the key/value is there
+    let value = view.val(b"smoke").await.unwrap();
+    assert_eq!(value, Some(b"test".to_vec()));
```

Comment:
Should only define the key and value once then use that definition everywhere. 

```suggestion
    let key = b"smoke";
    let value = b"test";

    // insert a single key/value
    let batch_put = BatchOp::Put {
        key,
        value,
    };
    let proposal: Arc<firewood::db::Proposal> = db.propose(vec![batch_put]).await?.into();
    proposal.commit().await?;

    // ensure the latest hash is different
    let latest = db.root_hash().await?;
    assert_ne!(empty_hash, latest);

    // fetch the view of the latest
    let view = db.revision(latest).await.unwrap();

    // check that the key/value is there
    assert_eq!(view.val(key).await.unwrap().unwrap(), value);
```
you can compare `Vec<T>` with `&[T]`.

Reply:
Please do this before merge